ldap_jwt_auth/core/maintenance.py

```diff
@@ -0,0 +1,49 @@
+"""
+Module for handling maintenance requests
+"""
+
+import json
+
+from pydantic import ValidationError
+from ldap_jwt_auth.core.exceptions import InvalidMaintenanceFileFormat, MissingMaintenanceFile
+from ldap_jwt_auth.core.schemas import MaintenanceState, ScheduledMaintenanceState
+
+class Maintenance:
+    """
+    Class for managing maintenance requests.
+    """
+
+    def get_maintenance(self) -> MaintenanceState:
+        """
+        Return a schema for maintenance state of ims
+
+        :return: Maintenance state schema
+        :raises InvalidFileFormat: If the maintenance state file is incorrectly formatted
+        """
+        try:
+            with open("maintenance/maintenance.json", "r", encoding='utf-8') as file:
+                data = json.load(file)
+            maintenance: MaintenanceState = MaintenanceState(**data)
+            return maintenance
+        except IOError as exc:
+            print(exc)
```

Comment:
I think we need to catch a different exception here, I will have a look. Also `print` is not needed.

Reply:
@MatteoGuarnaccia5 Looks like `IOError` was merged into `OSError` in Python 3.3. `OSError` is a subclass of `Exception` and will catch all of these:
![image](https://github.com/ral-facilities/ldap-jwt-auth/assets/45173816/1162f979-c23f-4fe9-9b20-16ad11aea43e)

`json.load` can throw